src/main/java/dl/news/portal/web/config/oauth2/OauthAuthenticationSuccessHandler.java

```diff
@@ -0,0 +1,40 @@
+package dl.news.portal.web.config.oauth2;
+
+import dl.news.portal.domain.dto.UserDto;
+import dl.news.portal.domain.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.oauth2.provider.OAuth2Authentication;
+import org.springframework.security.web.authentication.SavedRequestAwareAuthenticationSuccessHandler;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Map;
+
+public class OauthAuthenticationSuccessHandler extends SavedRequestAwareAuthenticationSuccessHandler {
+    @Autowired
+    private UserService userService;
+
+    @Override
+    public void onAuthenticationSuccess(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse, Authentication authentication) throws IOException, ServletException {
+        super.onAuthenticationSuccess(httpServletRequest, httpServletResponse, authentication);
+        OAuth2Authentication oAuth2Authentication = (OAuth2Authentication) authentication;
+        String username = oAuth2Authentication.getName();
+        if (!userService.findByUsername(username).isPresent()) {
+            UserDto userDto = createNewUserDto(oAuth2Authentication.getUserAuthentication());
+            userService.createUser(userDto);
+        }
+    }
+
+    private UserDto createNewUserDto(Authentication authentication) {
+        String email = (String) ((Map) authentication.getDetails()).get("email");
+        UserDto userDto = new UserDto();
+        userDto.setUsername(authentication.getName());
+        userDto.setPassword("password");
```

Comment:
password - 'password'? weird thing. think of something else for this. even null would be better. Or it will ok to create separate table for yousers from oAuth server, and link this table to general user table.

Reply:
I split common and special for authenticate user's information and create separate tables for these.